plugins/quickbooks-online/src/src/Facade/QuickBooksFacade.php

```diff
@@ -283,12 +330,36 @@ private function getQBClient(DataService $dataService, int $ucrmClientId)
 
     private function createQBLineFromItem(DataService $dataService, array $item, int $qbIncomeAccountNumber)
     {
-        return $dataService->Add(Item::create([
-            'Name' => sprintf('%s (UCRMID-%s)', $item['label'], $item['id']) ,
-            'Type' => 'Service',
-            'IncomeAccountRef' => [
-                'value' => $qbIncomeAccountNumber
-            ],
-        ]));
+        try {
+            $response = $dataService->Add(
+                Item::create(
+                    [
+                        'Name' => sprintf('%s (UCRMID-%s)', $item['label'], $item['id']),
+                        'Type' => 'Service',
+                        'IncomeAccountRef' => [
+                            'value' => $qbIncomeAccountNumber,
+                        ],
+                    ]
+                )
+            );
+
+            $this->throwExceptionOnErrorResponse($dataService);
+
+            return $response;
+        } catch (\Exception $exception) {
+            $this->logger->error(
+                sprintf(' Item ID: %s export failed with error %s.', $item['id'], $exception->getMessage())
+            );
+        }
+    }
+
+    private function throwExceptionOnErrorResponse(DataService $dataService): void
+    {
+        /** @var FaultHandler $error */
+        if ($error = $dataService->getLastError()) {
+            $xml = new \SimpleXMLElement($error->getResponseBody());
+            $message = $xml->Fault->Error->Detail;
```

Comment:
some `isset` check would be nice here + try/catch on the XML, what if it's not XML, we crash?